module/zfs/spa.c

```diff
@@ -3261,6 +3317,7 @@ spa_add_l2cache(spa_t *spa, nvlist_t *config)
 			    ZPOOL_CONFIG_VDEV_STATS, (uint64_t **)&vs, &vsc)
 			    == 0);
 			vdev_get_stats(vd, vs);
+			spa_update_l2cache_stats_ex(vd, l2cache[i]);
```

Comment:
This feels really clunky, surely we can use `vdev_config_generate_stats()` for this.  Right?